hydra-node/src/Hydra/Chain/Direct/Handlers.hs

```diff
@@ -67,40 +73,42 @@ import System.IO.Error (userError)
 data LocalChainState m = LocalChainState
   { getLatest :: STM m ChainStateAt
   , pushNew :: ChainStateAt -> STM m ()
-  , rollback :: ChainPoint -> STM m ChainStateAt
+  , rollback :: ChainPoint -> STM m (ChainStateType Tx)
+  , history :: STM m (ChainStateHistory Tx)
   }
 
 -- | Initialize a new local chain state with given 'ChainStateAt' (see also
 -- 'initialChainState').
 newLocalChainState ::
   MonadSTM m =>
-  ChainStateAt ->
+  ChainStateHistory Tx ->
   m (LocalChainState m)
-newLocalChainState chainStateAt = do
-  tv <- newTVarIO chainStateAt
+newLocalChainState chainState = do
+  tv <- newTVarIO chainState
   pure
     LocalChainState
-      { getLatest = readTVar tv
+      { getLatest = getLatest tv
       , pushNew = pushNew tv
       , rollback = rollback tv
+      , history = readTVar tv
       }
  where
+  getLatest tv = do
+    chainStateHistory <- readTVar tv
+    return (currentState chainStateHistory)
```

Comment:
Minor: We use `pure` over `return` throughout the codebase. Not sure if you have a strong opinion on this, I don't. We could avoid altogether her by using `fmap`.